### PR TITLE
add homework8

### PR DESCRIPTION
**Pull request by DaniilKorinnyi**
Add homework8

### CODE REVIEW

File: homework8/task1.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from datetime import datetime
+
+
+def decorator(func):
+  def wrap():
```
Comment (klymenok) [inline homework8/task1.py:5]: Use *args and **kwargs here to make in works with functions that take arguments. And pass these arguments to the func() call

File: homework8/task3.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class EqualSignContext:
+    def __init__(self):
+        self.sign = "="
+
+    def __enter__(self):
+        print(self.sign*10)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        print(self.sign*10)
```
Comment (klymenok) [inline homework8/task3.py:9]: Move this line to the end of the method. And put the return statement after (move it outside the `if` block

**Review by klymenok — CHANGES_REQUESTED** [review]

**Review by DaniilKorinnyi — COMMENTED** [review]
Some corrections in task 1 and 3

File: homework8/task3.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    def __init__(self):
+        self.sign = "="
+
+    def __enter__(self):
+        print(self.sign*10)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if exc_val:
+            print(f"Error occured: {exc_val}")
+            print(self.sign * 10)
```
Comment (klymenok) [inline homework8/task3.py:11]: The line should be printed in any case. Move it outside the print block

**Review by klymenok — CHANGES_REQUESTED** [review]

**Review by DaniilKorinnyi — COMMENTED** [review]
Corrected the mistakes.

File: homework8/task3.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        if exc_val:
+            print(f"Error occured: {exc_val}")
+        return True
+
+a = EqualSignContext()
+with a:
+    print("This code will be executed")
+    x = 1 / 0
+    print("This code will not be executed")
+print(a.sign * 10)
```
Comment (klymenok) [inline homework8/task3.py:18]: By `move this like outsite if block` I mean exactly moving outside a block but not outside the whole class. Move it back to `__exit__` method:
```
    def __exit__(self, exc_type, exc_val, exc_tb):
        if exc_val:
            print(f"Error occured: {exc_val}")
        print(a.sign * 10)
        return True

**Review by klymenok — CHANGES_REQUESTED** [review]

File: homework8/task3.py (hunk 4 above)
Comment (DaniilKorinnyi) [reply homework8/task3.py:18]: I tried to do in such way, but it didn't work. Now, I've realised in what problem was. Pardon me)

**Review by klymenok — APPROVED** [review]
Looks good!

**Pull request merged**